build-scripts/build_all_guides.py

```diff
@@ -90,7 +91,8 @@ def main():
         worker.daemon = True
         worker.start()
 
-    queue.join()
+    for worker in workers:
+        worker.join()
```

Comment:
shouldn't we also join the queue?

Reply:
If all workers exit and the queue has not joined, then we risk running forever. However, you are correct that the logic in the workers called `queue.task_done()`; this wouldn't work under this system --and would leave the workers running--so I changed it to `break`.

Reply:
This might behave a little more optimally now. I also cleared the queue and left the `queue.task_done()` and the `break`.